lib/Backends/NNPI/NNPIOptions.h

```diff
@@ -18,166 +18,306 @@
+  }
+
 class NNPIBackendOptions : public NNPIOptions {
 public:
-  explicit NNPIBackendOptions();
-  /// Dump the status of all variables.
-  virtual std::string dumpStatus() const override;
-  /// Lists the supported options (name=>description).
-  static llvm::StringMap<std::string> getSupportedOptions();
   /// Compile to ICE-T if true, ICE-Ref otherwise.
-  bool useIceT;
-  /// Enable execution on device (useIceT_ and !inferOnDevice_ will compile but
+  DECLARE_NNPI_OPTION(useIceT, bool, "UseIceT", "Use inference api",
+                      "USE_ICE_T", "0");
+  /// Enable execution on device (useIceT and !inferOnDevice will compile but
   /// not execute inference).
-  bool inferOnDevice;
+  DECLARE_NNPI_OPTION(inferOnDevice, bool, "InferOnDevice",
+                      "Enable execution on device (UseIceT and !InferOnDevice "
+                      "will compile but not execute inference).",
+                      "USE_INF_API", "0");
   /// Setting this to true will log the status of all variables at backend
   /// creation.
-  bool showVars;
+#ifdef NDEBUG
+  DECLARE_NNPI_OPTION(showVars, bool, "ShowVars",
+                      "Setting this to true will log the status of all "
+                      "variables at backend creation.",
+                      "NNPI_SHOW_VARS", "0");
+#else
+  DECLARE_NNPI_OPTION(showVars, bool, "ShowVars",
+                      "Setting this to true will log the status of all "
+                      "variables at backend creation.",
+                      "NNPI_SHOW_VARS", "1");
+#endif
+  NNPIBackendOptions() {
+    INIT_NNPI_OPTIONS(useIceT, llvm::StringMap<std::string>());
+    INIT_NNPI_OPTIONS(inferOnDevice, llvm::StringMap<std::string>());
+    INIT_NNPI_OPTIONS(showVars, llvm::StringMap<std::string>());
+  }
 
-protected:
-  static std::map<std::string, std::pair<std::string, std::string>>
-  supportedOptionsMap();
+  virtual std::string getOptionsName() const override {
+    return "Backend Options";
+  };
 };
 
-/// This class holds all environment variable knobs for the NNPI compilation.
 class NNPICompilationOptions : public NNPIOptions {
 public:
-  /// Compilation paramters used as defaults (if exists) that may be overriden
-  /// by environment variables.
-  NNPICompilationOptions(const std::map<std::string, std::string> *parameters);
-  /// Dump the status of all variables.
-  virtual std::string dumpStatus() const override;
-  /// Lists the supported options (name=>description).
-  static llvm::StringMap<std::string> getSupportedOptions();
-  /// Compile to ICE-T if true, ICE-Ref otherwise - this variable cab be
-  /// verridden by inferOnDevice_.
-  bool useIceT;
-  /// Enable execution on device (useIceT_ and !inferOnDevice_ will compile but
+  /// Compile to ICE-T if true, ICE-Ref otherwise.
+  DECLARE_NNPI_OPTION(useIceT, bool, "UseIceT", "Use inference api",
+                      "USE_ICE_T", "0");
+  /// Enable execution on device (useIceT and !inferOnDevice will compile but
```

Comment:
Same as other comment -- can we expand on descriptions of interactions between `useIceT` and `inferOnDevice`?